src/Perlang.Interpreter/Typing/TypeResolver.cs

```diff
@@ -447,6 +463,16 @@ private static ITypeReference GreaterType(ITypeReference leftTypeReference, ITyp
                     // of the language, the cost in terms of added complexity might not be worth it.
                     return null;
 
+                case TypeCode.Object:
+                    if (type == typeof(BigInteger))
+                    {
+                        // Note: this is insanely wrong and will likely cause more problems than the ones it intends to
+                        // solve...
+                        return Convert.ToDouble(Decimal.MaxValue);
+                    }
+
+                    return null;
+
```

Comment:
I think we should look a bit more into this before merging. Can we provoke any obvious errors caused by the simplistic logic above? :thinking: Perhaps in a unit test or so.

Reply:
Added a comment for now, resolving.